.github/workflows/build-and-test-pipeline.yml

```diff
@@ -20,16 +18,14 @@ jobs:
     - run: docker run todo-app:test tests
     - run: docker run -e API_KEY=${{ secrets.TRELLO_API_KEY }} -e API_TOKEN=${{ secrets.TRELLO_API_TOKEN }} -e ORGANISATION_ID=${{ secrets.ORGANISATION_ID }} -e SECRET_KEY=${{ secrets.SECRET_KEY }} todo-app:test e2e_tests
   publish: 
-    name: Publish and Deploy to Heroku
+    name: Publish and Deploy to Azure
     runs-on: ubuntu-latest
     needs: build
     if: github.event_name == 'push' && github.ref == 'refs/heads/main'
     steps:
       - uses: actions/checkout@v2
       - run: echo ${{ secrets.DOCKERHUB_PASSWORD }} | docker login -u ${{ secrets.DOCKERHUB_USERNAME }} --password-stdin
-      - run: docker build --target production --tag ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:$GITHUB_SHA .
-      - run: docker push ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:$GITHUB_SHA
-      - run: docker tag ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:$GITHUB_SHA registry.heroku.com/devops-kris-todo-app/web
-      - run: heroku container:login
-      - run: docker push registry.heroku.com/devops-kris-todo-app/web
-      - run: heroku container:release web -a devops-kris-todo-app
+      - run: docker build --target production --tag ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest .
+      - run: docker push ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest
+      - run: curl -dH -X POST ${{ secrets.AZURE_WEBHOOK_URL }}
```

Comment:
One nice addition here can be adding the [--fail](https://curl.se/docs/manpage.html#-f) flag, as otherwise curl reports success as long as a successful connection was made, even if the response reports a failure (e.g. a status code > 400)